src/otx/recipe/keypoint_detection/openvino_model.yaml

```diff
@@ -0,0 +1,53 @@
+model:
+  class_path: otx.core.model.keypoint_detection.OVKeypointDetectionModel
+  init_args:
+    label_info: 19
+    model_name: rtm_pose_tiny
+    model_type: "keypoint_detection"
+    async_inference: true
+    use_throughput_mode: true
+
+engine:
+  task: KEYPOINT_DETECTION
+  device: cpu
+
+callback_monitor: val/PCK
+
+data: ../_base_/data/keypoint_detection.yaml
+overrides:
+  reset:
+    - data.train_subset.transforms
+    - data.val_subset.transforms
+    - data.test_subset.transforms
+
+  data:
+    stack_images: false
+    train_subset:
+      batch_size: 1
+      num_workers: 2
+      transforms:
+        - class_path: otx.core.data.transform_libs.torchvision.GetBBoxCenterScale
+        - class_path: otx.core.data.transform_libs.torchvision.TopdownAffine
+          init_args:
+            input_size: $(input_size)
+            is_numpy_to_tvtensor: true
+
+    val_subset:
+      batch_size: 1
+      num_workers: 2
+      transforms:
+        - class_path: otx.core.data.transform_libs.torchvision.GetBBoxCenterScale
+        - class_path: otx.core.data.transform_libs.torchvision.TopdownAffine
+          init_args:
+            input_size: $(input_size)
+            is_numpy_to_tvtensor: true
+
+    test_subset:
+      batch_size: 64
+      num_workers: 2
+      transforms:
+        - class_path: otx.core.data.transform_libs.torchvision.GetBBoxCenterScale
+        - class_path: otx.core.data.transform_libs.torchvision.TopdownAffine
```

Comment:
Are these transformations necessary for ov inference? Isn't there any post-processing for predicted key points in model api?

Reply:
I agree with OVModel must be quick substitutes to torch ones.
But I think OTX must expect post-processed outputs for validation and test because what we and users really want is processed ones, not raw predictions.
And det, iseg, and vpm seem to use post-processed predictions for metric calculation or returning outputs (I don't know why semantic segmentation and keypoint detection don't adapt this pipeline).
Could you share your opinion about this @kprokofi @harimkang?

Reply:
The same problem applies to hierarchical classification: we have to disregard postprocessing coming from MAPI, because otherwise we'd have to reproduce the same postprocessing in torch. Dete Also, we're using the same prediction entities for OV and torch models. If the outputs of OV and torch are different, while the same entities are used, that'd rise questions from user's side, let alone neither of output formats is documented.
If we think about usecases for real inference (the results are supposedly utilized outside OTX), I won't say OTX batch entities are somehow convenient. The real strong reason to prefer OTX inference over MAPI is CUDA, but we don't provide meaningful inference results in that scenario and do not even target that.

Reply:
But for detection case, post-processing for predicted bounding boxes is already being used, isn't it?
https://github.com/openvinotoolkit/model_api/blob/master/model_api/python/model_api/models/detection_model.py#L78-L124

Current OTX's models are based on mmlab models and as far as I know those models usually have their own post-processing pipeline like below detection usecase.
https://github.com/openvinotoolkit/training_extensions/blob/b99770a95d2b5ce8c72ec85534a4d690428eb0de/src/otx/algo/detection/heads/base_head.py#L313-L374

Current OTX detection models don't do pre-processing for ground truths during validation or inference considering above things.
(the arg `transform_bbox` of `otx.core.data.transform_libs.torchvision.Resize` is set to False as default)
https://github.com/openvinotoolkit/training_extensions/blob/b99770a95d2b5ce8c72ec85534a4d690428eb0de/src/otx/recipe/_base_/data/detection.yaml#L37-L77

I agree with your opinion about using unified prediction entity for both otx and ov models, but I think the predictions must be post-processed one. If no, I think it's like losing one of MAPI's core features in OTX.

Reply:
I see that for some of the models predictions made by torch and OV models could be close to the ready-to use one, but that's nothing more than a coincidence. OTX defines prediction format by defining an evaluation procedure, and that's it. Previously, I proposed introduction of two modes of prediction in OTX: one for metric computation, and one for generating user-friendly predictions. My proposal was declined. We can discuss this once again, but the current arrangement is that inference is done by MAPI, not OTX.
For kp task to resolve the issue you mention, we can file a ticket to Wonju, so he refactors torch & evaluation to support prediction & evaluation of postprocessed keypoints coordinates, then the OV model can be upgraded accordingly.